pkg/storaged/crypto-keyslots.jsx

```diff
@@ -139,19 +139,17 @@ function passphrase_change(block, key, new_passphrase, old_passphrase) {
                          { superuser: true, err: "message" }).input(old_passphrase + "\n" + new_passphrase + "\n");
 }
 
-function passphrase_remove(block, passphrase) {
+function slot_remove(block, slot, passphrase) {
     var dev = decode_filename(block.Device);
-    return cockpit.spawn(["cryptsetup", "luksRemoveKey", dev],
-                         { superuser: true, err: "message" }).input(passphrase);
-}
+    const cmd = ["cryptsetup", "luksKillSlot", dev, slot.toString()];
+    if (passphrase === false)
+        cmd.splice(2, 0, "-q");
 
-/* Generic slot operations
- */
+    const spawn = cockpit.spawn(cmd, { superuser: true, err: "message" });
+    if (passphrase !== false)
+        spawn.input(passphrase + "\n");
```

Comment:
Note that luksKillSlot will succeed if `passphrase` is any valid passphrase it doesn't need to be the one in the given slot.
So we still have the issue that people might accidentally remove the wrong slot.

Reply:
Man page: " Except running  in  batch-mode (-q) a remaining passphrase must be supplied"  Does that maybe even mean that one must provide a passphrase that is in some other slot than the one that is about to be removed, as a proof that you still know how to unlock this thing after removing the slot?

Reply:
So the thing is: Now you are removing slot by typing its passphrase. If you click on 'Remove slot 3' but write down passphrase for slot 2 it will remove slot 2. See  #15773

With this change we will remove only the slot that you select from UI and you need to write _any other_ passphrase (not the one that you are removing).

Reply:
>  Does that maybe even mean that one must provide a passphrase that is in some other slot than the one that is about to be removed, as a proof that you still know how to unlock this thing after removing the slot?

Exactly.

Reply:
> With this change we will remove only the slot that you select from UI and you need to write any other passphrase (not the one that you are removing).

Oh, I always understood the dialog the other way around: "please type the passphrase you want to remove".